src/main/java/org/json/JSONObject.java

```diff
@@ -253,7 +253,11 @@ public JSONObject(JSONTokener x) throws JSONException {
             switch (x.nextClean()) {
             case ';':
             case ',':
-                if (x.nextClean() == '}') {
+                c = x.nextClean();
+                if (c == 0) {
```

Comment:
maybe change to `if (x.end()) {`

`end` would match the expected use of the Tokener API better.

Reply:
Yes, that's cleaner!